Sources/SmartInsights/SmartInsightsProviding.swift

```diff
@@ -0,0 +1,16 @@
+//
+//  SmartInsightsProviding.swift
+//
+//  Created by Antoine van der Lee on 10/02/2022.
+//  Copyright © 2019 WeTransfer. All rights reserved.
+//
+
+import Foundation
+
+public protocol SmartInsightsProviding {
+
+    /// Allows parsing the given chapter and read Smart Insights out of it.
+    /// - Parameter `chapter`: The `DiagnosticsChapter` to use for reading out insights.
+    /// - Returns: An optional collection of smart insights derived from the chapter.
+    func smartInsights(for chapter: DiagnosticsChapter) -> [SmartInsightProviding]?
```

Comment:
What's `[]` vs `nil` here? How do we discern them?

Reply:
Do you think there's a reason to discern them? From my perspective it allows returning `nil` instead of requiring to create an empty array if there's no insights to return.

Reply:
Right, but I, as a user of this library, would wonder what the difference is. Which now I can guess, or try and figure out. But as there is no discernible difference down the line, I'd opt for returning an empty array versus an optional one. But up to you. :)

Reply:
I agree with Bas. In general I think optional arrays are something to be avoided as the difference between an empty array and `nil` is usually not needed and just confusing.